src/Core/Managed/Shared/OperationTelemetryExtensions.cs

```diff
@@ -16,9 +17,13 @@ public static class OperationTelemetryExtensions
         /// </summary>
         /// <param name="telemetry">Telemetry item object that calls this extension method.</param>
         public static void Start(this OperationTelemetry telemetry)
-        {
-            var startTime = DateTimeOffset.UtcNow;
-            telemetry.Timestamp = startTime;
+        {            
+            telemetry.Timestamp = DateTimeOffset.UtcNow;
+
+            // Begin time is used internally for calculating duration of operation at the end call,
+            // and hence is stored using higher precision Clock.
+            // Stopwatch.GetTimestamp() is used (instead of ElapsedTicks) as it is thread-safe.
+            telemetry.BeginTimeInTicks = Stopwatch.GetTimestamp();
```

Comment:
Original code was using a globally shared StopWatch instance and reading Ticks using stopwatch's Elapsed property
which is not thread-safe as per MSDN doc below.
https://msdn.microsoft.com/en-us/library/system.diagnostics.stopwatch(v=vs.110).aspx

So I switched to thread-safe static method StopWatch.GetTimeStamp(), which eliminated the need for having a Globally shared instance of stopwatch in product code.